File: backend/src/main/java/com/smartcampus/service/EventService.java
package com.smartcampus.service;

import com.smartcampus.dto.EventDto;
import com.smartcampus.model.EventStatus;

import java.util.List;

/**
 * Service interface for Event management.
 */
public interface EventService {
    // Admin Functions
    EventDto createEvent(EventDto eventDto);
    EventDto updateEvent(String eventId, EventDto eventDto);
    void deleteEvent(String eventId);
    List<EventDto> getAllEventsForAdmin();
    void cancelEvent(String eventId);

    // User Functions
    List<EventDto> getAllEvents(boolean includePast);
    EventDto getEventById(String eventId);
    void registerForEvent(String eventId);
    void cancelRegistration(String eventId);
    List<EventDto> getUserRegisteredEvents();
    
    // Utility
    void updateParticipantCount(String eventId);
    void sendEventReminders();
}

File: frontend/src/services/eventService.js
import api from './api';

/**
 * Service for Event-related API calls.
 */
export const getEvents = (includePast = false) => api.get(`/events?includePast=${includePast}`);
export const getEventById = (id) => api.get(`/events/${id}`);
export const registerForEvent = (id) => api.post(`/events/${id}/register`);
export const cancelRegistration = (id) => api.delete(`/events/${id}/cancel`);
export const getMyEvents = () => api.get('/events/my-events');
export const getMyEventsCount = () => api.get('/events/my-count');

// Admin Functions
export const getAllEventsAdmin = () => api.get('/events/admin');
export const createEvent = (data) => api.post('/events', data);
export const updateEvent = (id, data) => api.put(`/events/${id}`, data);
export const deleteEvent = (id) => api.delete(`/events/${id}`);
